src/common.jl

```diff
@@ -18,14 +14,38 @@ function get_total_q(l::PSY.StandardLoad)
            PSY.get_impedance_reactive_power(l)
 end
 
-function get_total_p(l::PSY.ExponentialLoad)
-    return PSY.get_active_power(l)
+"""
+Return the reactive power limits that should be used in power flow calculations and PSS/E
+exports. Redirects to `PSY.get_reactive_power_limits` in all but special cases.
+"""
+get_reactive_power_limits_for_power_flow(gen::PSY.Device) =
+    PSY.get_reactive_power_limits(gen)
+
+function get_reactive_power_limits_for_power_flow(gen::PSY.RenewableNonDispatch)
+    val = PSY.get_reactive_power(gen)
+    return (min = val, max = val)
 end
 
-function get_total_q(l::PSY.ExponentialLoad)
-    return PSY.get_reactive_power(l)
+"""
+Return the active power limits that should be used in power flow calculations and PSS/E
+exports. Redirects to `PSY.get_active_power_limits` in all but special cases.
+"""
+get_active_power_limits_for_power_flow(gen::PSY.Device) = PSY.get_active_power_limits(gen)
+
+get_active_power_limits_for_power_flow(::PSY.Source) = (min = -Inf, max = Inf)
+
+function get_active_power_limits_for_power_flow(gen::PSY.RenewableNonDispatch)
+    val = PSY.get_active_power(gen)
+    return (min = val, max = val)
 end
 
+get_active_power_limits_for_power_flow(gen::PSY.RenewableDispatch) =
+    (min = 0.0, max = PSY.get_rating(gen))
+
+# TODO verify whether this is the correct behavior for Storage, (a) for redistribution and (b) for exporting
```

Comment:
open an issue for this. We don't know what the correct approach is yet.

Reply:
-> https://github.com/NREL-Sienna/PowerFlows.jl/issues/88